src/main/java/com/google/devtools/build/lib/shell/JavaSubprocessFactory.java

```diff
@@ -172,6 +172,8 @@ public Subprocess create(SubprocessBuilder params) throws IOException {
                       .put(
                           StringEncoding.internalToPlatform(key),
                           StringEncoding.internalToPlatform(value)));
+    } else {
+      builder.environment().putAll(params.getClientEnv());
```

Comment:
Why does `params.getClientEnv()` not need to be converted? If it's already platform, doesn't that violate the principle that all Bazel strings should remain internal up until the point where they're passed into an external API?

Reply:
Thanks for catching this, I simply forgot to update this.